File: go-client-sdk/pkg/models/operations/postsubscriptionssubscriptionidunschedulependingplanchanges.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PostSubscriptionsSubscriptionIDUnschedulePendingPlanChangesPathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscription_id"`
}

type PostSubscriptionsSubscriptionIDUnschedulePendingPlanChangesRequest struct {
	PathParams PostSubscriptionsSubscriptionIDUnschedulePendingPlanChangesPathParams
}

type PostSubscriptionsSubscriptionIDUnschedulePendingPlanChangesResponse struct {
	ContentType  string
	StatusCode   int
	Subscription *shared.Subscription
}

File: typescript-client-sdk/src/sdk/models/operations/postsubscriptionssubscriptionidunschedulependingplanchanges.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class PostSubscriptionsSubscriptionIdUnschedulePendingPlanChangesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscription_id" })
  subscriptionId: string;
}

export class PostSubscriptionsSubscriptionIdUnschedulePendingPlanChangesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostSubscriptionsSubscriptionIdUnschedulePendingPlanChangesPathParams;
}

export class PostSubscriptionsSubscriptionIdUnschedulePendingPlanChangesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  subscription?: shared.Subscription;
}